chore(macros/HTTPSidebar): update Observatory links

chrisdavidmills opened the pull request:
<!--
  Thanks for taking the time to submit a pull request (PR)!
  Please provide enough information so that others can review your changes.

  The sections below are mandatory.
  If you don't follow this template, your PR will very likely be closed.

  Before submitting the PR, please make sure the following is done:
  1. Read the Community Participation Guidelines: https://www.mozilla.org/about/governance/policies/participation/
  2. Ensure that there is an open issue for the problem you're solving, or create it first: https://github.com/mdn/yari/issues/new/choose
  3. Fork the repository and create your branch from `main`.
  4. Run `yarn` in the repository root.
  5. Make sure to sign all your commits: https://docs.github.com/authentication/managing-commit-signature-verification/signing-commits
-->

## Summary

In https://github.com/mdn/content/pull/33793, I am adding the security docs that accompany Mozilla Observatory to MDN. I noticed that after this work is merged and Observatory is moved over to MDN, the HTTP docs sidebar will then contain out-of-date links.

This PR fixes that sidebar to link to the correct new locations of the Observatory tool and associated docs.

This shouldn't be merged until the above two items are done.

<!--
  Please reference the issue you're solving here.
  Example: Fixes #1234.
-->

### Problem

<!--
  Explain what problem the PR resolves in 1-3 sentences.
-->

### Solution

<!--
  Explain how your PR solves this problem in 1-3 sentences.
  Please mention alternative solutions you have discarded, if any.
-->

---

## Screenshots

| Before | After |
|--------|--------|
| <img width="286" alt="image" src="https://github.com/mdn/yari/assets/495429/6c385dad-f0f1-4f40-8ce4-08762bd16332"> | <img width="286" alt="image" src="https://github.com/mdn/yari/assets/495429/aafdbed2-c26a-4c87-a0c5-c78b0c107ac6"> | 

---

## How did you test this change?

<!--
  Did you change anything else (other than the user interface)?
  If not, you can remove this section.
  If yes, please explain how you verified that your PR solves the problem you wanted to solve.
-->

chrisdavidmills committed:
Update Observatory links in HTTPsidebar

chrisdavidmills committed:
remove trailing slash

caugner reviewed (changes requested):
The changes don't look right, as if a VSCode extension reformatted the file in a weird way. This not only makes it hard to review the changes, but appears to have caused syntax errors that cause the tests to fail. Can you please take a look?

PS: Afaik there is no formatter for EJS, so you might have to reset and start from scratch.

chrisdavidmills commented:
> The changes don't look right, as if a VSCode extension reformatted the file in a weird way. This not only makes it hard to review the changes, but appears to have caused syntax errors that cause the tests to fail. Can you please take a look?
> 
> PS: Afaik there is no formatter for EJS, so you might have to reset and start from scratch.

@caugner ah man, that really sucks. I didn't even notice it had done that. I will have another go at this tomorrow.

caugner committed:
Merge branch 'main' into update-http-observatory-links

caugner committed:
fix(macros/HTTPSidebar): revert formatting + redo changes

1. Going forward, we avoid localizing titles in macros.
2. The Observatory will only be available in /en-US/.

caugner committed:
Merge branch 'observatory' into update-http-observatory-links

caugner committed:
fix(macros/HTTPSidebar): update new security guides slug

caugner committed:
fix(macros/HTTPSidebar): use classic link for Observatory

caugner commented:
@chrisdavidmills Fixed the PR, and added screenshots. Will merge this into the `observatory` branch now.
